### ISSUE TITLE
The implemetation of the focal loss make me confused.

### ISSUE THREAD

**Issue by chengcchn** [open-mmlab/mmdetection#812]
The implemetation of the focal loss make me confused. Why calculate "p * log(1-p)" in such a complicated way. As the pic show, term2 = p^gamma * log(1-p). The calculate way was seperate into two part. If logits>=0, log(1-p) = -logits-log(1+e^(-logits)), else, log(1-p) = -log(1+e^logits).
![Snipaste_2019-06-16_16-35-11](https://user-images.githubusercontent.com/38886481/59563908-90b58f00-9072-11e9-9d27-6d0ae39bf1b7.png)

**Comment by hellock** [comment]
Yes it can be simplified. This implementation is translated from the python implementation.

**Comment by chengcchn** [comment]
@hellock Thanks!
